cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_lin.cpp

```diff
@@ -0,0 +1,456 @@
+    if(move_lin.rotate_only)
+    {
+        Se_max = std::max(std::fabs(Se), Se_max);
+        Se_max = std::max(std::fabs(Se_roll), Se_max);
+        Se_max = std::max(std::fabs(Se_pitch), Se_max);
+        Se_max = std::max(std::fabs(Se_yaw), Se_max);
+    }
+    else    // Otherwise set the linear-path as Se_max
+    {
+        Se_max = Se;
+    }
+
+    // Calculate the Ramp Profile Parameters
+    if (move_lin.profile.vel > sqrt(std::fabs(Se_max) * move_lin.profile.accl))
+    {
+        move_lin.profile.vel = sqrt(std::fabs(Se_max) * move_lin.profile.accl);
+    }
+    tb = move_lin.profile.vel / move_lin.profile.accl;
+    te = (std::fabs(Se_max) / move_lin.profile.vel) + tb;
+    tv = te - tb;
+
+//ToDo: steps_tb is int! Do we need ceil() or floor()?
+    // Interpolationsteps for every timesequence
+    steps_tb = (double)tb      / t_ipo_;
+    steps_tv = (double)(tv-tb) / t_ipo_;
+    steps_te = (double)(te-tv) / t_ipo_;
+
+//ToDo: What's the purpose of this? division by t_ipo_ followed by multiplication with t_ipo_?
+    // Reconfigure timings wtih t_ipo_
+    tb = steps_tb * t_ipo_;
+    tv = (steps_tb + steps_tv) * t_ipo_;
+    te = (steps_tb + steps_tv + steps_te) * t_ipo_;
+
+    //Convert to RPY
+    double roll, pitch, yaw;
+    tf::Quaternion q;
+    tf::quaternionMsgToTF(move_lin.start.orientation, q);
+    tf::Matrix3x3(q).getRPY(roll, pitch, yaw);
+
+    // Calculate the paths
+    if(!generateRampPath(linear_path, move_lin.profile.vel, move_lin.profile.accl, Se_max, (steps_tb + steps_tv + steps_te)))
+    {
+        ROS_WARN("Error while Calculating linear_path");
+        return false;
+    }
+    if(!generateRampPathWithTe(roll_path, te, move_lin.profile.accl, Se_roll, (steps_tb + steps_tv + steps_te), roll))
+    {
+        ROS_WARN("Error while Calculating roll_path");
+        return false;
+    }
+    if(!generateRampPathWithTe(pitch_path, te, move_lin.profile.accl, Se_pitch, (steps_tb + steps_tv + steps_te), pitch))
+    {
+        ROS_WARN("Error while Calculating pitch_path");
+        return false;
+    }
+    if(!generateRampPathWithTe(yaw_path, te, move_lin.profile.accl, Se_yaw, (steps_tb + steps_tv + steps_te), yaw))
+    {
+        ROS_WARN("Error while Calculating yaw_path");
+        return false;
+    }
+
+    // Resize the path vectors
+    unsigned int max_steps = 0;
+    max_steps = std::max((unsigned int)linear_path.size(), max_steps);
+    max_steps = std::max((unsigned int)roll_path.size(), max_steps);
+    max_steps = std::max((unsigned int)pitch_path.size(), max_steps);
+    max_steps = std::max((unsigned int)yaw_path.size(), max_steps);
+    
+    linear_path.resize(max_steps, linear_path.back());
+    roll_path.resize(max_steps, roll_path.back());
+    pitch_path.resize(max_steps, pitch_path.back());
+    yaw_path.resize(max_steps, yaw_path.back());
+
+    // Put the interpolated paths into the path_matrix
+    path_matrix[0] = linear_path;
```

Comment:
Better use enum or defines instead of naked numbers -> allows same meaning in several methods.